pennylane/ops/channel.py

```diff
@@ -365,6 +365,92 @@ def _kraus_matrices(cls, *params):
         return [K0, K1, K2, K3, K4]
 
 
+class PauliError(Channel):
+    r"""PauliError(operators, p, wirelist, wires)
+    Arbitrary number qubit, arbitrary Pauli error channel.
+
+    This class models corresponding Kraus matrices to a series of arbitrary Pauli operators on
+    an arbitrary amount of wires. The operators are supplied in a string. The according wires are
+    supplied in an array called wirelist. The probability with which the channel is applied is
+    supplied as a float. Each wire can take one Pauli operator.
+
+    This channel is modelled by the following Kraus matrices:
+
+    .. math::
+        K_0 = \sqrt{1-p} * I
+
+    .. math::
+        K_1 = \sqrt{p} * (K_{w0} \otimes (K_{w1} \otimes (\dots K_{w_{max}})))
+
+    Where :math:`I` is the Identity,
+    and :math:`\otimes` denotes the Kronecker Product,
+    and :math:`K_wi` denotes the Kraus matrix corresponding to the operator acting on wire i,
+    and :math:`p` denotes the probability with which the channel is applied
+
+    **Warning:**
+    Because the Kraus Matrices scale exponentially to the amount of wires, their dimension blow up
+    fast. Use with caution!
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 3
+
+    Args:
+        operators (str): The Pauli operators acting on the specified (groups of) wires
+        p (float): The probability of the operator being applied
+        wirelist (Sequence[int]): The wires the channel acts on
+        wires (Sequence[int]): The wires the channel acts on
+    """
+
+    num_params = 3
+    num_wires = AnyWires
+    par_domain = "L"
+
+    ops = {
+        "X": np.array([[0, 1], [1, 0]]),
+        "Y": np.array([[0, -1j], [1j, 0]]),
+        "Z": np.array([[1, 0], [0, -1]]),
+    }
+
+    @classmethod
+    def _kraus_matrices(cls, *params):
+        operators, p, wirelist = params[0], params[1], params[2]
+
+        # check if the specified operators are legal
+        if not all(c in "XYZ" for c in operators):
+            raise ValueError("The specified operators need to be either of 'X', 'Y' or 'Z'")
+
+        # check if probabilities are legal
+        if not 0.0 <= p <= 1.0:
+            raise ValueError("p must be between [0,1]")
+
+        # check if the number of operators matches the number of wires
+        if len(operators) != len(wirelist):
+            raise ValueError("The number of operators must match the number of wires")
+
+        max_q = max(wirelist)
```

Comment:
Would we like to query the length of the wire list here instead? :thinking: 

E.g., if we have `wirelist = [0, 4, 8]` for a device with 9 wires which have indices of consecutive integers from 0 (0, 1, ...8), then `max_q` would be `8` in this example. However, we'd only like to apply the channel on 3 qubits.

This likely also ties in with the question of whether or not to include the identity for wires that are "enclosed" by the wires PauliError acts on. Based on the logic below, it seems that we'd like to recreate Kraus matrices by considering identities too. Would we actually need to consider identities? :slightly_smiling_face: 

Another thing to note: wires can have custom labels (as per the Wires class).
```suggestion
        max_q = len(wirelist)
```